demo/scripts/controls/sidePane/apiPlayground/vtable/VTablePane.tsx

```diff
@@ -1,364 +1,776 @@
+                onClick={() => {
+                    formatTable(
+                        editor,
+                        setBandedRow(this.state.vtable.table),
+                        this.state.vtable.table
+                    );
+
+                    this.forceUpdate();
+                }}>
+                Banded Row
+            </button>
+        );
+    }
+
+    private renderFormatTableButton(
+        text: string,
+        format: TableFormat,
+        editor: IEditor
+    ): JSX.Element {
+        return (
+            <button
+                className={styles.button}
+                onClick={() => {
+                    saveTableInfo(this.state.vtable.table, format);
+                    formatTable(editor, format, this.state.vtable.table);
+                    this.forceUpdate();
+                }}>
+                {text}
+            </button>
+        );
+    }
+
+    private onCustomizeFormat = () => {
+        const format = createTableFormat(
+            this.bgColor.current.value || undefined,
+            this.topBorderColor.current.value || undefined,
+            this.bottomBorderColor.current.value || undefined,
+            this.verticalBorderColor.current.value || undefined
+        );
+
+        this.state.vtable.applyFormat(format);
+        this.forceUpdate();
+    };
+
+    private onWriteBack = () => {
+        const editor = this.props.getEditor();
+        editor.addUndoSnapshot(() => {
+            const vtable = this.state.vtable;
+            const td = vtable.getCurrentTd();
+            vtable.writeBack();
+            editor.focus();
+            editor.select(td, PositionType.Begin);
+        });
+        this.createVTable();
+    };
+}
+
+function createTableFormat(
+    bgColor?: string,
+    topBorder?: string,
+    bottomBorder?: string,
+    verticalBorder?: string,
+    bandedRows?: boolean,
+    bandedColumns?: boolean,
+    headerRow?: boolean,
+    firstColumn?: boolean,
+    borderFormat?: TableBorderFormat,
+    bgColorEven?: string,
+    bgColorOdd?: string,
+    bgColumnColorEven?: string,
+    bgColumnColorOdd?: string,
+    headerRowColor?: string
+): TableFormat {
+    return {
+        bgColor: bgColor,
+        topBorderColor: topBorder,
+        bottomBorderColor: bottomBorder,
+        verticalBorderColor: verticalBorder,
+        bandedRows: bandedRows,
+        bgColorEven: bgColorEven,
+        bgColorOdd: bgColorOdd,
+        bandedColumns: bandedColumns,
+        bgColumnColorEven: bgColumnColorEven,
+        bgColumnColorOdd: bgColumnColorOdd,
+        headerRow: headerRow,
+        headerRowColor: headerRowColor,
+        firstColumn: firstColumn,
+        tableBorderFormat: borderFormat,
+    };
+}
+
+function setHeaderRow(table: HTMLTableElement): Partial<TableFormat> {
+    const format = getTableFormatInfo(table);
+    format.headerRow = !format.headerRow;
+    if (format.headerRow) {
+        format.headerRowColor = format.bottomBorderColor;

```

Comment:
I think you should add the header color always when creating the table format.
So you don't need to modify the TableFormat here. and just toggle the 'header row' state.

If you dont want to apply the header row when setting a predefined table styles, set the headerRow state by default as false.